charms/knative-serving/src/charm.py

```diff
@@ -95,16 +96,27 @@ def _get_custom_images(self):
             )
         return custom_images
 
-    def _on_install(self, _):
+    def _main(self, event):
```

Comment:
nitpick: I don't think `_main` makes intuitive sense as a name for the method. I assume you changed it because it can now be deferred and thus emitted on any event, but something more descriptive may be more fitting. Maybe `_try_apply_and_change_status` since that's what it actually does
```suggestion
    def _main(self, event):
```

Reply:
In other charms we name the common event handler as `_main` or even `_event`, but you are right that the naming is less explicit. I could rename it, but then in the future if we decide to just use it as the main event handler, we'd need to change it again. I'm okay with either option, what do you think?

Reply:
Going for consistency is the most sensible option here, so let's keep it as `_main` to be consistent with our other charms